runtime/oti/VMHelpers.hpp

```diff
@@ -2048,13 +2049,60 @@ class VM_VMHelpers
 #endif /* JAVA_SPEC_VERSION > 11 */
 	}
 
+#if JAVA_SPEC_VERSION >= 19
+/*
+ *	If low tagging failed (or not even attempted) due to either
+ *
+ *   a carrier thread winning to mount, we don't need to do anything, since it will be compensated by pre/post mount actions
+ *   another GC thread winning to scan, again don't do anything, and let the winning thread do the work, instead
+ */
+	static VMINLINE bool
+	tryEnterConcurrentContinuationMonitor(J9VMContinuation *continuation)
+	{
+		volatile uintptr_t *localAddr = (volatile uintptr_t *) &continuation->carrierThreadAndConcurrentlyScanned;
+		uintptr_t oldValue = *localAddr;
+		uintptr_t retValue = 0;
+
+		if (0 == oldValue) {
+			/* atomically low tag ('set', not 'or') continuation->carrierThreadAndConcurrentlyScanned */
+			if ((retValue = VM_AtomicSupport::lockCompareExchange(localAddr, oldValue, J9_GC_CONTINUATION_CONCURRENTSCANNING)) == oldValue) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	static VMINLINE void
+	exitConcurrentContinuationMonitor(J9VMContinuation *continuation)
+	{
+		volatile uintptr_t *localAddr = (volatile uintptr_t *) &continuation->carrierThreadAndConcurrentlyScanned;
+
+		uintptr_t oldValue = VM_AtomicSupport::bitAnd(localAddr, ~(uintptr_t)J9_GC_CONTINUATION_CONCURRENTSCANNING);
+		if (J9_ARE_ANY_BITS_SET(oldValue, ~(uintptr_t)J9_GC_CONTINUATION_CONCURRENTSCANNING)) {
+			J9VMThread *carrierThread = (J9VMThread *) *localAddr;
```

Comment:
Not sure if it's really the same, but I feel more comfortable with getting carrier thread from oldValue , rather then re-reading from current carrierThreadAndConcurrentlyScanned.

And it would probably be nice to call it oldCarrierThreadAndConcurrentlyScanned